diff --git a/discover.h b/discover.h
index 6c594b7..8b0e6bf 100644
--- a/discover.h
+++ b/discover.h
@@ -52,7 +52,7 @@ private:
     eConnectTimeoutMs = 1500,  // in milliseconds
     eProbeTimeoutMs   = 2000,  // in milliseconds
     eProbeIntervalMs  = 60000, // in milliseconds
-    eCleanupTimeoutMs = 124000 // in milliseoonds
+    eCleanupTimeoutMs = 1240000 // in milliseoonds
   };
   static cSatipDiscover *instanceS;
   static size_t HeaderCallback(char *ptrP, size_t sizeP, size_t nmembP, void *dataP);
